compiler/rustc_target/src/asm/arm64ec.rs

```diff
@@ -0,0 +1,185 @@
+use std::fmt;
```

Comment:
Unfortunately, I could not maintain the approach of using the aarch64 implementation to support asm for arm64ec.

If I wanted to just reject `p*`/`ffr`, it worked with the previous file structure (https://github.com/rust-lang/rust/pull/131332/commits/955d8c63925435b976f2985f64a8c079833522e5), but I couldn't figure out how to reject z0-z15, which are defined as super registers for v0-v15.
(Filter in def_regs! (like `% restricted_for_arm64ec`) could not be used for this purpose because it did not know what the original register name was. Also, z* could not be defined as a separate clobber-only register class because it [can accept values on aarch64](https://godbolt.org/z/Pvor9dvq3).)